src/Microsoft.ML.AutoML/DatasetDimensions/ReservoirSampledDataView.cs

```diff
@@ -0,0 +1,321 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.ML.Data;
+
+namespace Microsoft.ML.AutoML
+{
+    /// <summary>
+    /// A data view that is a reservoir sample of a passed-in, parent data view.
+    /// All of the sampled data is held IN MEMORY, so be careful not to make the
+    /// sample size too large.
+    /// </summary>
+    internal class ReservoirSampledDataView : IDataView
+    {
+        private readonly IDataView _originalData;
+        private readonly Random _random;
+        private readonly object[] _sampledColumns;
+        private readonly long _sampleSize;
+
+        private bool _initialized = false;
+        private long _originalDataRowCount;
+
+        public ReservoirSampledDataView(IDataView data, long sampleSize)
+        {
+            _originalData = data;
+            _random = new Random();
```

Comment:
May want to expose a seed value as an optional parameter. Having a stable sample should make unit tests easier, both in the reservoir sample tests, and the statistics unit tests.

Reply:
Added